$OpenBSD: patch-apps_snmpps_c,v 1.1 2020/01/03 22:34:35 sthen Exp $

OpenBSD snmpd doesn't support HOST-RESOURCES-MIB::hrSWRunPerfMem /
HOST-RESOURCES-MIB::hrSWRunPerfCPU and returns SNMP_NOSUCHOBJECT.
Just bail rather than null-deref'ing.

Index: apps/snmpps.c
--- apps/snmpps.c.orig
+++ apps/snmpps.c
@@ -365,11 +365,13 @@ collect_perf(netsnmp_session *ss, struct hrSWRunTable 
         proc.hrSWRunStatus = *vlp2->val.integer;
 
         vlp2 = vlp2->next_variable;
-        if (vlp2->type == SNMP_NOSUCHINSTANCE) goto next;
+        if (vlp2->type == SNMP_NOSUCHINSTANCE ||
+                    vlp2->type == SNMP_NOSUCHOBJECT) goto next;
         proc.hrSWRunPerfCPU = *vlp2->val.integer;
 
         vlp2 = vlp2->next_variable;
-        if (vlp2->type == SNMP_NOSUCHINSTANCE) goto next;
+        if (vlp2->type == SNMP_NOSUCHINSTANCE ||
+                    vlp2->type == SNMP_NOSUCHOBJECT) goto next;
         proc.hrSWRunPerfMem = *vlp2->val.integer;
 
         count++;
